presto-hive/src/main/java/com/facebook/presto/hive/security/ranger/RangerBasedAccessControl.java

```diff
@@ -0,0 +1,516 @@
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    public void checkCanCreateSchema(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, String schemaName)
+    {
+    }
+
+    /**
+     * Check if identity is allowed to drop the specified schema in this catalog.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    public void checkCanDropSchema(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, String schemaName)
+    {
+        SchemaTableName schemaTableName = new SchemaTableName(schemaName, schemaName);
+        if (!checkAccess(identity, schemaTableName, null, HiveAccessType.DROP)) {
+            denyCreateSchema(schemaTableName.toString(), format("Access denied - User [ $s ] does not have is not have [DROP] " +
+                    "privilege on [ %s ] ", identity.getUser(), schemaTableName.getSchemaName()));
+        }
+    }
+
+    /**
+     * Check if identity is allowed to execute SHOW SCHEMAS in a catalog.
+     * <p>
+     * NOTE: This method is only present to give users an error message when listing is not allowed.
+     * The {@link #filterSchemas} method must handle filter all results for unauthorized users,
+     * since there are multiple way to list schemas.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    @Override
+    public void checkCanShowSchemas(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context)
+    {
+    }
+
+    /**
+     * Filter the list of schemas to those visible to the identity.
+     */
+    @Override
+    public Set<String> filterSchemas(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, Set<String> schemaNames)
+    {
+        Set<String> allowedSchemas = new HashSet<>();
+        Set<String> groups = getGroupsForUser(identity.getUser());
+
+        for (String schema : schemaNames) {
+            if (rangerAuthorizer.authorizeHiveResource(schema, null, null, HiveAccessType.SELECT.toString(), identity.getUser(), groups)) {
+                allowedSchemas.add(schema);
+            }
+        }
+        return allowedSchemas;
+    }
+
+    /**
+     * Check if identity is allowed to create the specified table in this catalog.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    @Override
+    public void checkCanCreateTable(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, SchemaTableName tableName)
+    {
+        if (!checkAccess(identity, tableName, null, HiveAccessType.CREATE)) {
+            denyCreateTable(tableName.toString(), format("Access denied - User [ $s ] does not have is not have [CREATE] " +
+                    "privilege on [ %s ] ", identity.getUser(), tableName.getSchemaName()));
```

Comment:
Fix extraInfo as per suggestion above and please do the same for all instances below as well